src/Sulu/Bundle/ContentBundle/Search/Metadata/StructureProvider.php

```diff
@@ -198,7 +200,18 @@ public function getMetadata(Metadata $documentMetadata, StructureMetadata $struc
         }
 
         if ($class->isSubclassOf(ResourceSegmentBehavior::class)) {
-            $indexMeta->setUrlField($this->factory->createMetadataField('resourceSegment'));
+            $field = $this->factory->createMetadataField('resourceSegment');
+            if ($class->isSubclassOf(RedirectTypeBehavior::class)) {
+                $field = new Expression(
+                    sprintf(
+                        '(object.getRedirectType() === %s ? (object.getRedirectTarget() ? object.getRedirectTarget().getResourceSegment() : "") : (object.getRedirectType() === %s ? object.getRedirectExternal() : object.getResourceSegment()))',
```

Comment:
Naive question: Isn't it possible to split this in two different expressions? But if yes, then the check on `object.getRedirectType` could  be omitted.

I also think `: ""` could simply be omitted, since according to the [Symfony Documentation](http://symfony.com/doc/current/components/expression_language/syntax.html) :

```
foo ? 'yes' (equal to foo ? 'yes' : '')
```

If it is not possible to split (although when looking at it again, it should be possible, since you are already using different fields for `ResourceSegmentBehavior`and non-`ResourceSegmentBehavior` documents) I would do it like that:

```
(object.getRedirectType() === %s
    ? (object.getRedirectTarget() ? object.getRedirectTarget().getResourceSegment())
    : (object.getRedirectType() === %s
        ? object.getRedirectExternal() 
        : object.getResourceSegment()
    )
)
```

I'm also not sure if falling back to `object.getResourceSegment()` if the external link is not set is right...